web/src/lib/components/admin-page/settings/proxy/proxy-settings.svelte

```diff
@@ -0,0 +1,143 @@
+<script lang="ts">
+  import {
+    notificationController,
+    NotificationType,
+  } from '$lib/components/shared-components/notification/notification';
+  import { handleError } from '$lib/utils/handle-error';
+  import { api, ProxyProtocol, SystemConfigProxyDto } from '@api';
+  import { isEqual } from 'lodash-es';
+  import { fade } from 'svelte/transition';
+  import ConfirmDisableLogin from '../confirm-disable-login.svelte';
+  import SettingButtonsRow from '../setting-buttons-row.svelte';
+  import SettingSwitch from '../setting-switch.svelte';
+  import SettingSelect from '../setting-select.svelte';
+  import SettingInputField, { SettingInputFieldType } from '../setting-input-field.svelte';
+
+  export let proxyConfig: SystemConfigProxyDto; // this is the config that is being edited
+
+  let savedConfig: SystemConfigProxyDto;
+  let defaultConfig: SystemConfigProxyDto;
+
+  async function getConfigs() {
+    [savedConfig, defaultConfig] = await Promise.all([
+      api.systemConfigApi.getConfig().then((res) => res.data.proxy),
+      api.systemConfigApi.getDefaults().then((res) => res.data.proxy),
+    ]);
+  }
+
+  let isConfirmOpen = false;
+  let handleConfirm: (value: boolean) => void;
+
+  const openConfirmModal = () => {
+    return new Promise((resolve) => {
+      handleConfirm = (value: boolean) => {
+        isConfirmOpen = false;
+        resolve(value);
+      };
+      isConfirmOpen = true;
+    });
+  };
+
+  async function saveSetting() {
+    try {
+      const { data: current } = await api.systemConfigApi.getConfig();
+
+      if (!current.oauth.enabled && current.proxy.enabled && !proxyConfig.enabled) {
+        const confirmed = await openConfirmModal();
+        if (!confirmed) {
+          return;
+        }
+      }
```

Comment:
Why is there OAuth stuff here? Why do you need a confirm modal here?

Reply:
It was my template for the proxy settings, I didn't finish at all this part